src/stan/lang/ast_def.cpp

```diff
@@ -284,6 +284,16 @@ namespace stan {
     void function_signatures::add_unary(const::std::string& name) {
       add(name, DOUBLE_T, DOUBLE_T);
     }
+    void function_signatures::add_unary_vectorized(const::std::string&
+    name) {
+      for (size_t i = 0; i < 8; ++i) {
+        add(name, expr_type(DOUBLE_T, i), expr_type(INT_T, i));
```

Comment:
In the unary vectorize code in `math`, the function to deal with the base case of int returns double: 
(https://github.com/stan-dev/math/blob/1f2794953e5199717afd87bf40dbbe1f7165ce20/stan/math/prim/mat/vectorize/apply_scalar_unary.hpp#L101-L120). So, if the function is using the unary vectorize code without any modification, the return type for any int types will be double.